nannyml/drift/ranking.py

```diff
@@ -200,3 +260,118 @@ def rank(
         if only_drifting:
             ranking = ranking.loc[ranking['number_of_alerts'] != 0, :]
         return ranking
+
+@Ranker.register('correlation')
+class CorrelationRanking(Ranking):
+    """Ranks features according to the correlation of their drift results and
+    absolute performance change from mean reference performance.
+    """
+
+
+    def fit(
+        self,
+        performance_results: Union[
+            CBPEResults, DLEResults, PerformanceCalculationResults
+        ],
+    ):
+        """
+        Use performance results from the reference period in order to be able to compare drift
+        results with absolute performance changes from the mean performance results during the reference period.
+
+        Parameters
+        ----------
+        performance_results : Performance Estimation or Performance Calculation results filtered to
+            the performance metric we want to compare against drift results. Additionally results need
+            to be filtered to the reference period only.
+            Can be an instance of:
+            nml.performance_estimation.confidence_based.results.Result,
+            nml.performance_estimation.direct_loss_estimation.result.Result,
+            nml.performance_calculation.result.Result
+        """
+
+        # User should filter
+        metrics = performance_results.metrics
+        if len(metrics) > 1:
+            raise ValueError("Only one metric should be present in performance_results used to fit CorrelationRanking.")
+        
+        self.metric = metrics[0]
+        self.mean_perf_value = performance_results.to_df().loc[:, (self.metric.column_name, 'value')].mean()
+
+
+    def rank(
+        self,
+        univariate_results: UnivariateResults,
+        performance_results: Union[
+            CBPEResults, DLEResults, PerformanceCalculationResults
+        ],
+        only_drifting: bool = False,
+
+    ):
+
+        # Let's do some checks that results objects are appropriate.
+        _univ_index = univariate_results.to_df().loc[:, ('chunk', 'chunk', 'start_index')]
+        _perf_index = performance_results.to_df().loc[:, ('chunk', 'start_index')]
+
+        if not _univ_index.equals(_perf_index):
```

Comment:
Maybe split off the checks into a separate function?

Reply:
Done with cdf8029fbe5446f6eabb694b3823de60ab070989